src/core/plugin/PluginController.cpp

```diff
@@ -1,54 +1,106 @@
-#ifdef ENABLE_PLUGINS
-    Settings* settings = control->getSettings();
-    std::vector<std::string> pluginEnabled = StringUtils::split(settings->getPluginEnabled(), ',');
-    std::vector<std::string> pluginDisabled = StringUtils::split(settings->getPluginDisabled(), ',');
+namespace {
+auto load_awailable_plugins_from(fs::path const& path, Control* control) -> std::vector<std::unique_ptr<Plugin>> {
+    if (!fs::is_directory(path)) {
+        g_info("Skipping Plugin path, it is not a directory: %s", path.string().c_str());
+        return {};
+    }
+    g_info("Loading plugins from: %s", path.string().c_str());
 
+    std::vector<std::unique_ptr<Plugin>> returner;
     try {
         for (auto const& f: fs::directory_iterator(path)) {
             const auto& pluginPath = f.path();
-            auto plugin = std::make_unique<Plugin>(control, pluginPath.filename().string(), pluginPath);
-            if (!plugin->isValid()) {
-                g_warning("Error loading plugin \"%s\"", f.path().string().c_str());
-                continue;
+            try {
+                auto plugin = std::make_unique<Plugin>(control, pluginPath.filename().string(), pluginPath);
+                if (!plugin || !plugin->isValid()) {
+                    g_warning("Error loading plugin \"%s\"", f.path().string().c_str());
+                    continue;
+                }
+                plugin->setEnabled(plugin->isDefaultEnabled());
+                returner.emplace_back(std::move(plugin));
+            } catch (std::exception const& e) {
+                g_warning("Error loading plugin \"%s\": %s", f.path().string().c_str(), e.what());
             }
+        }
+    } catch (fs::filesystem_error const& e) {
+        g_warning("Could not open plugin dir: \"%s\": %s", path.string().c_str(), e.what());
+    }
+    return returner;
+}
 
-            if (plugin->isDefaultEnabled()) {
-                plugin->setEnabled(!(std::find(pluginDisabled.begin(), pluginDisabled.end(), plugin->getName()) !=
-                                     pluginDisabled.end()));
-            } else {
-                plugin->setEnabled(std::find(pluginEnabled.begin(), pluginEnabled.end(), plugin->getName()) !=
-                                   pluginEnabled.end());
-            }
+auto emplace_sorted_if_not_exists(std::vector<std::unique_ptr<Plugin>>& plugins, std::unique_ptr<Plugin> plugin)
+        -> void {
+    auto const& path = plugin->getPath();
+    auto endit = std::end(plugins);
+    auto it = std::find_if(std::begin(plugins), endit,
+                           [&path](auto const& p) { return fs::canonical(p->getPath()) == fs::canonical(path); });
+    if (it != endit) {
+        return;
+    }
+    plugins.emplace_back(std::move(plugin));
+    std::sort(begin(plugins), end(plugins), [](auto const& lhs, auto const& rhs) {
+        auto tieer = [](auto&& p) { return std::tie(p->getName(), p->getPath()); };
+        return tieer(lhs) < tieer(rhs);
+    });
```

Comment:
Could this be improved by using `std::lower_bound` instead of inserting and sorting? Also, why not `*lhs < *rhs`?

Reply:
Replaced by a Comparator Class